es-vue-base/src/lib-components/EsFileUpload.vue

```diff
@@ -0,0 +1,230 @@
+            const totalFileSize = this.files.reduce((acc, file) => acc + file.size, 0);
+            if (totalFileSize > this.maxFileSize * 1000000) {
+                this.$emit('fileSizeError');
+                return false;
+            }
+            return true;
+        },
+        readFilesIntoUrl(files) {
+            this.fileUrls = [];
+            files
+                .forEach((file) => {
+                    const fileReader = new FileReader();
+                    fileReader.onload = () => {
+                        this.$emit('fileDataRead', {
+                            name: file.name, type: file.type, size: file.size, data: fileReader.result,
+                        });
+                    };
+                    fileReader.readAsDataURL(file);
+                });
+        },
+        onDrop(event) {
+            // The user has dropped files onto the component. We have to apply the same logic as if they had
+            // selected the files from the file picker which limits the file types to the ones specified in the
+            // fileTypes prop.
+            this.active = false;
+            this.files = [];
+
+            // Use DataTransferItemList interface to access the file(s)
+            const invalidFiles = [...event.dataTransfer.items].filter((item) => item.kind !== 'file');
+
+            const dataTransfersAsFiles = [...event.dataTransfer.items]
+                .filter((item) => item.kind === 'file')
+                .map((item) => item.getAsFile());
+
+            const validFiles = dataTransfersAsFiles
+                .filter((file) => this.fileTypes.includes(file.type));
+            const fileErrors = dataTransfersAsFiles
+                .filter((file) => !this.fileTypes.includes(file.type))
+                .concat(invalidFiles)
+                .map((item) => item.name);
+            if (fileErrors.length > 0) {
+                this.$emit('fileTypeError', fileErrors);
+            }
+
+            this.files = validFiles;
+            if (this.files.length > 0 && this.fileSizeValid()) {
+                this.$emit('readyToUpload', this.files.length);
+                this.readFilesIntoUrl(validFiles);
+            }
+        },
+        openFilePicker() { this.$refs.fileInput.$el.childNodes[0].click(); },
+        upload() {
+            this.files.forEach((file, index) => {
+                this.uploadSingleFile(file, index);
+            });
+        },
+        uploadSingleFile(file, index) {
+            const config = {
+                headers: {
+                    'Content-Type': file.type,
+                },
+                onUploadProgress: (progressEvent) => {
+                    const percentCompleted = Math.round((progressEvent.loaded * 100) / progressEvent.total);
+                    this.$emit('uploadProgress', { fileName: file.name, percentCompleted });
+                    return percentCompleted;
+                },
+            };
```

Comment:
Although I like emitting the event; I think showing the progress of the upload to the user is a responsibility of this component.